src/producer/vector_producer.rs

```diff
@@ -0,0 +1,396 @@
+                        .map(|s| s + "\n")
+                        .collect::<String>();
+
+                    // Log the exact payload for debugging
+                    tracing::debug!("Payload being sent:\n{}", body);
+
+                    tracing::debug!("Sending request to Vector with body size {}", body.len());
+                    if let Err(e) = client
+                        .post(&endpoint)
+                        .header("Content-Type", "application/json")
+                        .body(body)
+                        .send()
+                        .await
+                    {
+                        tracing::error!(error = ?e, "Failed to send batch request to Vector");
+                    } else {
+                        tracing::debug!("Successfully sent batch request");
+                    }
+                    batch.clear();
+                }
+            }
+
+            // Send any remaining events in the batch
+            if !batch.is_empty() {
+                tracing::debug!("Sending final batch of {} events", batch.len());
+                let body = batch
+                    .iter()
+                    .filter_map(|json| String::from_utf8(json.clone()).ok())
+                    .collect::<Vec<String>>()
+                    .join("\n");
+
+                // Log the exact payload for debugging
+                tracing::debug!("Final payload being sent:\n{}", body);
+
+                tracing::debug!(
+                    "Sending final request to Vector with body size {}",
+                    body.len()
+                );
+                if let Err(e) = client
+                    .post(&endpoint)
+                    .header("Content-Type", "application/json")
+                    .body(body)
+                    .send()
+                    .await
+                {
+                    tracing::error!(error = ?e, "Failed to send final batch request to Vector");
+                } else {
+                    tracing::debug!("Successfully sent final batch request");
+                }
+            }
+
+            tracing::info!("Vector producer worker shutting down");
+        })
+    }
+}
+
+impl Drop for VectorResultsProducer {
+    fn drop(&mut self) {
+        // No longer need to handle shutdown in Drop since the worker is managed externally
+    }
+}
+
+impl ResultsProducer for VectorResultsProducer {
+    fn produce_checker_result(&self, result: &CheckResult) -> Result<(), ExtractCodeError> {
+        let json = serde_json::to_vec(result)?;
+        self.schema.validate_json(&json)?;
+
+        // Send the serialized result to the worker task
+        if self.sender.send(json).is_err() {
+            tracing::error!("Failed to send result to Vector worker - channel closed");
+            return Err(ExtractCodeError::VectorRequestStatusError(
+                reqwest::StatusCode::INTERNAL_SERVER_ERROR,
+            ));
```

Comment:
This feels strange. send might fail because the channel is closed why are we producing a status code internal server error? I would expect a different error type.